### PR TITLE
Feature/cinemachine

### PR DESCRIPTION
**Pull request by m4rtint**
- Added Cinemachine - https://www.youtube.com/watch?v=Gx9gZ9cfrys
- Added in split screen - https://www.youtube.com/watch?v=X6Ou8InMUbg
- Added in Camera Shake - https://www.youtube.com/watch?v=O2Pg8e2xwzg

### CODE REVIEW

File: Assets/Script/Data/Utilities.cs [hunk 1]
```diff
@@ -102,5 +102,55 @@ public static bool IsPlayStationController(InputDeviceStyle style)
                 style == InputDeviceStyle.PlayStationMove ||
                 style == InputDeviceStyle.PlayStationVita;
         }
+
+        /*
+         *    TODO : TEST AND FOLLOW THESE RULES
+            * this.m_twoPlayers.Add(new ScreenAttributes(0.5f, 1.0f, 0, 0));
```
Comment (mikeslgoh) [inline Assets/Script/Data/Utilities.cs]: Is this just a reminder for unit testing?

File: Assets/Script/Game/PlayerController.cs [hunk 2]
```diff
@@ -94,14 +97,18 @@ private void SetupCar(PlayerInformation playerInfo, Transform spawnPoint)
...
     }
 
-    private void CreateCamera()
+    private void SetupCamera()
     {
-        GameObject createdCamera = SplitScreenManager.SetupPlayerCamera(m_car.gameObject, PlayerNumber(), this.transform);
-        m_camera = createdCamera.GetComponent<Camera>();
+        int followCamLayer = LayerMask.NameToLayer(StoredStrings.PlayerCameraLayer(m_player));
+        m_VirtualCamera.layer = followCamLayer;
+        m_playerCamera.cullingMask |= (1 << followCamLayer);
```
Comment (mikeslgoh) [inline Assets/Script/Game/PlayerController.cs]: what does |= do?

File: ProjectSettings/TagManager.asset [hunk 3]
```diff
@@ -9,6 +9,7 @@ TagManager:
   - SpawnPoint
   - RoadConnection
   - WaitZone
+  - Gib
```
Comment (mikeslgoh) [inline ProjectSettings/TagManager.asset]: What is this tag for?

**Comment by mikeslgoh** [comment]
[Camera to match with perspective](https://trello.com/c/QMEXWRnB/305-camera-to-match-with-perspective)

**Comment by mikeslgoh** [comment]
[Add Camera shake](https://trello.com/c/m8Q4rOOY/307-add-camera-shake)

File: Assets/Script/Data/Utilities.cs (hunk 1 above)
Comment (m4rtint) [reply Assets/Script/Data/Utilities.cs]: Yes. and whoever does this unit test, they hopefully know what to look for.

File: Assets/Script/Game/PlayerController.cs (hunk 2 above)
Comment (m4rtint) [reply Assets/Script/Game/PlayerController.cs]: it's a bit mask. 
oldMask = oldMask | (1 << camLayer)

Comment (m4rtint) [reply Assets/Script/Game/PlayerController.cs]: OR bitmask

File: ProjectSettings/TagManager.asset (hunk 3 above)
Comment (m4rtint) [reply ProjectSettings/TagManager.asset]: Strange, i don't know why this wasn't saved. 

it's for the AI Vehicles. All AI Vehicles is tagged with Gib. if they're too close to another Gib, they will stop in their tracks.

It's for when there is a red light, and they have to wait for the car in front to go.

Comment (m4rtint) [reply ProjectSettings/TagManager.asset]: For more details if you want
https://docs.google.com/document/d/1HkpH4IA8Fnk78slJDWOK1Vt-4X8aaNW0YjRashw-YzU/edit?usp=sharing

File: Assets/CameraShake.cs [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+                // If Camera Shake effect is over, reset variables
+                virtualCameraNoise.m_AmplitudeGain = 0f;
+                ShakeElapsedTime = 0f;
+            }
+        }
+    }
+
+    private void OnCollisionEnter(Collision collision)
+    {
+        if (true ||collision.gameObject.CompareTag(StoredStrings.CarTag))
```
Comment (m4rtint) [inline Assets/CameraShake.cs]: Oh wait i need to fix this.

**Comment by m4rtint** [comment]
[Camera to match with perspective](https://trello.com/c/QMEXWRnB/305-camera-to-match-with-perspective)

File: Assets/Script/Game/Car/CameraShake.cs [hunk 5]
```diff
@@ -0,0 +1,68 @@
+﻿using Cinemachine;
+using UnityEngine;
+
+public class CameraShake : MonoBehaviour
+{
+    public float ShakeDuration = 0.3f;          // Time the Camera Shake effect will last
+    public float ShakeAmplitude = 1.2f;         // Cinemachine Noise Profile Parameter
+    public float ShakeFrequency = 2.0f;         // Cinemachine Noise Profile Parameter
```
Comment (Gursaanj) [inline Assets/Script/Game/Car/CameraShake.cs]: Why are these public? also should these be const variables 

Also the naming convention doesn't seem style copped haha

File: Assets/Script/Game/Car/CameraShake.cs [hunk 6]
```diff
@@ -0,0 +1,68 @@
+﻿using Cinemachine;
```
Comment (Gursaanj) [inline Assets/Script/Game/Car/CameraShake.cs]: Missing Header

File: Assets/Script/Game/Car/CarAudioSystem.cs [hunk 7]
```diff
@@ -0,0 +1,39 @@
+﻿using System.Collections;
```
Comment (Gursaanj) [inline Assets/Script/Game/Car/CarAudioSystem.cs]: Missing Header

File: Assets/Script/Game/Car/CameraShake.cs (hunk 5 above)
Comment (m4rtint) [reply Assets/Script/Game/Car/CameraShake.cs]: I took this from online. I'll Change it by tonight

**Pull request merged**